### ISSUE TITLE
Provide a GeoJSON output option

### ISSUE THREAD

**Issue by mvl22** [keepright/keepright#52]
https://github.com/keepright/keepright/blob/master/web/points.php#L198-L214

outputs the data as a TSV, e.g.:

http://keepright.ipax.at/points.php?lat=52.19638802775339&lon=0.1330569848822541&zoom=16&show_ign=1&show_tmpign=1&lang=en&ch=0,30,40,50,70,90,100,110,120,130,150,160,170,180,191,192,193,194,195,196,197,198,201,202,203,204,205,206,207,208,210,220,231,232,270,281,282,283,284,285,291,292,293,294,295,296,297,298,311,312,313,320,350,370,380,401,402,411,412,413

It would be useful to have an option to output the data as GeoJSON, with the different columns as properties, as that would enable easy embedding in another site.

Looking at the code this should just be a case of adding an option that basically just overrides lines 198-214 in the above PHP file.

**Comment by mueschel** [comment]
There is already an export script available at https://github.com/keepright/keepright/blob/master/web/export.php
Currently only gpx and rss are supported, but adding GeoJSON should be straight forward. Could you provide an example json file?

**Comment by mvl22** [comment]
Thanks. Yes, looking at the file, that should be easy to add as another type there.

There is an example of how a GeoJSON output should look:

http://www.cyclestreets.net/api/v2/photomap.locations/#example

The properties array of each 'feature' is basically where you would put the key-value pairs like ```error_name ```, ```error_type```, ```object_type```, etc. The critical thing is to ensure that the latitude and longitude are in the part of the hierarchy shown in the example.

So basically for each point:

- Put the latitude and longitude in geometry/coordinates/
- Remove them from the array of properties
- Loop through each remaining key-value pair in the array of properties, and put each within properties/ . Easiest way to do that in PHP is to do ```json_encode($properties)```

You can validate GeoJSON at http://geojsonlint.com/ , which will show the points on a map.

**Comment by mueschel** [comment]
On my testing page there is a new link in the bottom left corner. Could you check if this is what you want to have?
http://osm.mueschelsoft.de/keepright/report_map.php?zoom=14

**Comment by mvl22** [comment]
@mueschel Do you have the link sorry?

**Comment by mueschel** [comment]
Sorry, I thought you'll get a notification with the link. I don't want to have it publicly accessible. Could you send me a mail to jan@osm.mueschelsoft.de ?

**Comment by mvl22** [comment]
Thanks for that commit:
https://github.com/keepright/keepright/commit/cb8a73fb713fe51e89889eb4306adc332bdd6eaa

Thanks - the link you sent looks good.

Can this be pushed on to the live server now?

**Comment by mvl22** [comment]
I'm using the latest version of Chrome, but it's coming back as text/html for some reason, with the result that the javascript inspector says that ```responseJSON``` is empty.

Unfortunately, I think that the Content-Type at
https://github.com/keepright/keepright/blob/master/web/export.php#L146
is probably not well supported by browsers.

Either that or the header isn't being sent properly by PHP.

It might be worth seeing if:

    application/json; charset=UTF-8

would give more compatibility.

**Comment by mueschel** [comment]
The mime-type was not correctly forwarded on my test server. Now it should be ok.

**Comment by mvl22** [comment]
Thanks - the header looks a lot better.

But unfortunately I'm still getting this responseJSON = undefined problem:

![responsejson](https://cloud.githubusercontent.com/assets/361423/12695997/f3403948-c755-11e5-88d7-3b7da2ce99fb.png)

I suspect it's either the:

    header('Content-Disposition: attachment; filename="points.geojson"');

line, which is probably unnecessary anyway, as browsers have great plugins for viewing JSON directly these days, or that

    application/vnd.geo+json

line which perhaps jQuery doesn't like.

**Comment by mvl22** [comment]
Would it be possible to replace experimentally these two lines:

https://github.com/keepright/keepright/blob/master/web/export.php#L146-L147

with

```header('Content-Type: application/json; charset=utf-8');```

?

I think this will then work as expected.

**Comment by mueschel** [comment]
Done.

**Comment by mvl22** [comment]
Thanks - this does seem to be working now!

**Comment by mueschel** [comment]
The export was integrated in the main web page
